### PR TITLE
feat(compass): removes special handling for Screen events

### PR DESCRIPTION
**Pull request by leorossi**
## Description
As per this [Slack conversation](https://mongodb.slack.com/archives/G2L10JAV7/p1638280738434800) the special handling for `Screen` events should be removed, and keep the `track()` behavior.

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-status/lib/index.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
...
...ns:e})},clearSubview(){this.setState({subview:null,subviewStore:null,subviewActions:null})},enableModal(){this.setState({modal:!0})},disableModal(){this.setState({modal:!1})},hide(){this.disableProgressTrickle(),this.setState(this.getInitialState())},done(){this.disableProgressTrickle(),this.setState({progress:100,animation:!1,message:"",subview:null,subviewStore:null,subviewActions:null}),this.hide()}});function V(){return(V=Object.assign||function(e){for(var t=1;t<arguments.length;t++){var n=arguments[t];for(var r in n)Object.prototype.hasOwnProperty.call(n,r)&&(e[r]=n[r])}return e}).apply(this,arguments)}class O extends r.Component{render(){return i.a.createElement(s.StoreConnector,{store:D},i.a.createElement(y,V({actions:j},this.props)))}}k="StatusPlugin",(S="displayName")in(T=O)?Object.defineProperty(T,S,{value:k,enumerable:!0,configurable:!0,writable:!0}):T[S]=k;var P=O,E=n(5);const N={name:"Status",component:P};function M(e){e.registerRole("Application.Status",N),e.registerAction("Status.Actions",j),e.registerStore("Status.Store",D)}function C(e){e.deregisterRole("Application.Status",N),e.deregisterAction("Status.Actions"),e.deregisterStore("Status.Store")}t.default=P}])}));
```
Comment (addaleax) [inline packages/compass-status/lib/index.js:6]: These changes are entirely unrelated, right?

**Review by addaleax — APPROVED** [review]

File: packages/compass/src/main/telemetry.ts [hunk 2]
```diff
@@ -49,42 +55,32 @@ class CompassTelemetry {
       return;
     }
 
-    if (info.event === 'Screen') {
-      const { name, ...properties } = info.properties;
-      if (name === this.lastReportedScreen) {
-        return;
-      }
-
-      this.lastReportedScreen = name;
```
Comment (addaleax) [inline packages/compass/src/main/telemetry.ts:58]: I think we’d want to keep this bit of logic in place, actually?

File: packages/compass-status/lib/index.js (hunk 1 above)
Comment (leorossi) [reply packages/compass-status/lib/index.js:6]: Yeah, I don't know where they are coming from, will remove it shortly

**Review by addaleax — APPROVED** [review]
